packages/cli/test/integration/evaluation/test-definitions.api.test.ts

```diff
@@ -0,0 +1,296 @@
+import { TestDefinitionRepository } from '@/databases/repositories/test-definition.repository.ee';
+import { createAnnotationTags } from '@test-integration/db/executions';
+
+import { createUserShell } from './../shared/db/users';
+import { createWorkflow } from './../shared/db/workflows';
+import * as testDb from './../shared/test-db';
+import type { SuperAgentTest } from './../shared/types';
+import * as utils from './../shared/utils/';
+
+let authOwnerAgent: SuperAgentTest;
+let workflowUnderTest: WorkflowEntity;
+let evaluationWorkflow: WorkflowEntity;
+let otherWorkflow: WorkflowEntity;
+let ownerShell: User;
+let annotationTag: AnnotationTagEntity;
+const testServer = utils.setupTestServer({ endpointGroups: ['evaluation'] });
+
+beforeAll(async () => {
+	ownerShell = await createUserShell('global:owner');
+	authOwnerAgent = testServer.authAgentFor(ownerShell);
+});
+
+beforeEach(async () => {
+	await testDb.truncate(['TestDefinition', 'Workflow', 'AnnotationTag']);
+
+	workflowUnderTest = await createWorkflow({ name: 'workflow-under-test' }, ownerShell);
+	evaluationWorkflow = await createWorkflow({ name: 'evaluation-workflow' }, ownerShell);
+	otherWorkflow = await createWorkflow({ name: 'other-workflow' });
+	annotationTag = (await createAnnotationTags(['test-tag']))[0];
+});
+
+describe('GET /evaluation/test-definitions', () => {
+	test('should retrieve empty test definitions list', async () => {
+		const resp = await authOwnerAgent.get('/evaluation/test-definitions');
+		expect(resp.statusCode).toBe(200);
+		expect(resp.body.data.count).toBe(0);
+		expect(resp.body.data.testDefinitions).toHaveLength(0);
+	});
+
+	test('should retrieve test definitions list', async () => {
+		const newTest = Container.get(TestDefinitionRepository).create({
+			name: 'test',
+			workflow: { id: workflowUnderTest.id },
+		});
+		await Container.get(TestDefinitionRepository).save(newTest);
+
+		const resp = await authOwnerAgent.get('/evaluation/test-definitions');
+
+		expect(resp.statusCode).toBe(200);
+		expect(resp.body.data.count).toBe(1);
+		expect(resp.body.data.testDefinitions).toHaveLength(1);
+		expect(resp.body.data.testDefinitions[0].name).toBe('test');
+		expect(resp.body.data.testDefinitions[0].workflowId).toBe(workflowUnderTest.id);
+		expect(resp.body.data.testDefinitions[0].evaluationWorkflowId).toBe(null);
+	});
+
+	test('should retrieve test definitions list with pagination', async () => {
+		// Add a bunch of test definitions
+		const testDefinitions = [];
+
+		for (let i = 0; i < 15; i++) {
+			const newTest = Container.get(TestDefinitionRepository).create({
+				name: `test-${i}`,
+				workflow: { id: workflowUnderTest.id },
+			});
+			testDefinitions.push(newTest);
+		}
+
+		await Container.get(TestDefinitionRepository).save(testDefinitions);
+
+		// Fetch the first page
+		let resp = await authOwnerAgent.get('/evaluation/test-definitions?take=10');
+		expect(resp.statusCode).toBe(200);
+		expect(resp.body.data.count).toBe(15);
```

Comment:
We should probably also test that the returned test definitions are also correct (ids should be enough)

Reply:
I made an attempt, but could not quickly come up with a reliable way to add tests in a specific order (the retrieval is ordered by `createdAt` field), so decided to postpone it